lib/indico.js

```diff
@@ -12,36 +12,85 @@ var public_api = 'http://apiv2.indico.io/'
 indico.apiKey = false;
 indico.cloud = false;
 
-function service(name, privateCloud, apiKey) {
+function service(name, privateCloud, apiKey, apis) {
+  /* given a set of parameters, returns the proper url for the REST api*/
   if (privateCloud) {
     var prefix = 'http://' + privateCloud + '.indico.domains/';
-    var url = prefix + name.toLowerCase();
+    var url = prefix + name;
   } else {
-    var url = public_api + name.toLowerCase();
+    var url = public_api + name;
   }
   url = url + "?key=" + apiKey;
+  if (apis) {
+    url = url + "&apis=" + apis.join();
+  }
   return url
 }
 
+function formatResults(results, nameMap) {
+  var formatted_results = {};
+  for (result in results) {
+    if ('results' in results[result]) {
+      formatted_results[nameMap[result]] = results[result]['results'];
+    } else {
+      formatted_results[nameMap[result]] = results[result];
+    }
+  }
+  return formatted_results;
+}
 
-var make_request = function (name, data, config) {
-  var apiKey = settings.resolveApiKey(config, indico.apiKey);
-  var privateCloud = settings.resolvePrivateCloud(config, indico.cloud);
-
-  var body = {};
-  body['data'] = data;
-
+function addKeywordArguments(body, config) {
+  // pass along additional keyword args in JSON body
   for (var key in config) {
-    if (config.hasOwnProperty(key)
+    if (config.hasOwnProperty(key) 
```

Comment:
It's one of those "javascript good practices" sort of things. http://stackoverflow.com/questions/684672/loop-through-javascript-object

 I'll go ahead and remove though, since in this case it's not a problem.